js/main.js

```diff
@@ -42,4 +49,80 @@ $submitForm.addEventListener('submit', function (event) {
   data.entries.push(newObj);
   $source.setAttribute('src', './images/placeholder-image-square.jpg');
   $submitForm.reset($submitForm);
-});
+}
+
+// make a function that loads a dom tree
+function renderEntry(entry) {
+  var rowPadding = document.createElement('div');
+  rowPadding.className = 'row padding';
+  var columnHalf = document.createElement('div');
+  columnHalf.className = 'column-half';
+  rowPadding.appendChild(columnHalf);
+
+  var entryImg = document.createElement('div');
+  entryImg.className = 'entry-img';
+  columnHalf.appendChild(entryImg);
+
+  var imgSrc = document.createElement('img');
+  imgSrc.setAttribute('src', entry.photo);
+  entryImg.appendChild(imgSrc);
+
+  var entryText = document.createElement('div');
+  entryText.className = 'column-half';
+  rowPadding.appendChild(entryText);
+
+  var title = document.createElement('h2');
+  var titleText = document.createTextNode(entry.input);
+  title.appendChild(titleText);
+  entryText.appendChild(title);
+
+  var notes = document.createElement('p');
+  var notesText = document.createTextNode(entry.notes);
+  notes.appendChild(notesText);
+  entryText.appendChild(notes);
+  return rowPadding;
+}
+
+// use a loop to create a DOM tree for each entry in data model
+// loop throough the entries
+// append it to the page by appending the function to the ul
+// append it when DOMContentLoaded event is fired
+function appendRenderEntry(event) {
+  for (var i = 0; i < data.entries.length; i++) {
+    $ul.appendChild(renderEntry(data.entries[i]));
+  }
+}
+window.addEventListener('DOMContentLoaded', appendRenderEntry);
+
+// listen for a click event on the entries tab
+// if user presses entries tab, remove the hidden class from the data view entries element
+// add the hidden class to the form
+$entriesTab.addEventListener('click', clickEntries);
+function clickEntries(event) {
+  data.view = 'entries';
+  $entriesPage.setAttribute('class', '$entriesPage');
+  $entryForm.setAttribute('class', '$entryForm hidden');
+} if (data.entries.length > 0) {
+  $noEntry.setAttribute('class', 'no-entries hidden');
+} else {
+  $noEntry.setAttribute('class', 'no-entries');
+}
+// add an event listener to the new button
+// if user presses the button, remove hidden class from the form
+// add hidden class to the data view entries
+$newEntry.addEventListener('click', newEntry);
+function newEntry(event) {
+  data.view = 'entry-form';
+  $entryForm.setAttribute('class', '$entryForm');
+  $entriesPage.setAttribute('class', '$entriesPage hidden');
+}
```

Comment:
You would benefit from making a single function named something like `switchView` that receives an argument of the name of the view to switch to, and the function should show that view and hide all other views. That way, you won't need to make a new function for each new view created.

Reply:
I suggest looking back to `javascript-view-swapping` for some inspiration on how to tackle this 🤓